### PR TITLE
EZP-30251: Display the rendering of time & date when editing the Short/Long Date & Time format

### PR DESCRIPTION
**Pull request by adamwojs**
| Question      | Answer
| ------------- | ---
| Tickets       |  [EZP-30251](https://jira.ez.no/browse/EZP-30251) 
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Added support for matching user setting update form view based on various criteria. Example configuration:

```yml
ezpublish:
    system:
        admin_group:
            # ...
            user_settings_update_view:
                full:
                    ezplatform_admin_ui_datetime_format:
                        template: '@ezdesign/user/settings/update_datetime_format.html.twig'
                        match:
                            Identifier: [full_datetime_format, short_datetime_format]
                    ezplatform_admin_ui:
                        template: '@ezdesign/user/settings/update.html.twig'
                        match: true
```

Ref. https://github.com/ezsystems/ezplatform-admin-ui/pull/889

#### Checklist:
- [ ] Implement tests
- [X] Coding standards (`$ composer fix-cs`)

### CODE REVIEW

File: src/lib/View/UserSettings/Matcher/Identifier.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+use eZ\Publish\Core\MVC\Symfony\View\View;
+use EzSystems\EzPlatformUser\View\UserSettings\UpdateView;
+
+/**
+ * Match based on the user setting identifier.
+ */
+class Identifier implements ViewMatcherInterface
+{
+    /** @var string[] */
+    private $identifier = [];
```
Comment (mikadamczyk) [inline src/lib/View/UserSettings/Matcher/Identifier.php]:
```suggestion
    private $identifiers = [];
```

File: src/lib/View/UserSettings/Matcher/Identifier.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setMatchingConfig($matchingConfig): void
+    {
+        $this->identifier = (array)$matchingConfig;
```
Comment (mikadamczyk) [inline src/lib/View/UserSettings/Matcher/Identifier.php]:
```suggestion
        $this->identifiers = (array)$matchingConfig;
```

File: src/lib/View/UserSettings/Matcher/Identifier.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function match(View $view): bool
+    {
+        if (!$view instanceof UpdateView || $view->getUserSetting() === null) {
+            return false;
+        }
+
+        return in_array($view->getUserSetting()->identifier, $this->identifier);
```
Comment (mikadamczyk) [inline src/lib/View/UserSettings/Matcher/Identifier.php]:
```suggestion
        return in_array($view->getUserSetting()->identifier, $this->identifiers);
```

File: src/lib/View/UserSettings/UpdateView.php [hunk 4]
```diff
@@ -9,7 +9,36 @@
...
 use eZ\Publish\Core\MVC\Symfony\View\BaseView;
+use EzSystems\EzPlatformUser\UserSetting\UserSetting;
 
 class UpdateView extends BaseView
 {
+    /** @var \EzSystems\EzPlatformUser\UserSetting\UserSetting|null */
+    private $userSetting;
+
+    /**
+     * @return \EzSystems\EzPlatformUser\UserSetting\UserSetting
```
Comment (mikadamczyk) [inline src/lib/View/UserSettings/UpdateView.php]:
```suggestion
     * @return \EzSystems\EzPlatformUser\UserSetting\UserSetting|null
```

File: src/lib/View/UserSettings/UpdateViewProvider.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+use eZ\Publish\Core\MVC\Symfony\Matcher\MatcherFactoryInterface;
+use eZ\Publish\Core\MVC\Symfony\View\View;
+use eZ\Publish\Core\MVC\Symfony\View\ViewProvider;
+use Symfony\Component\HttpKernel\Controller\ControllerReference;
+
+class UpdateViewProvider implements ViewProvider
+{
+    /**
+     * @var \eZ\Publish\Core\MVC\Symfony\Matcher\MatcherFactoryInterface
```
Comment (mikadamczyk) [inline src/lib/View/UserSettings/UpdateViewProvider.php]: please, make one line comment

File: src/lib/View/UserSettings/UpdateViewProvider.php [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    public function getView(View $view)
+    {
+        if (($configHash = $this->matcherFactory->match($view)) === null) {
+            return null;
+        }
+
+        return $this->buildUpdateSettingView($configHash);
+    }
+
+    protected function buildUpdateSettingView(array $viewConfig): UpdateView
```
Comment (mikadamczyk) [inline src/lib/View/UserSettings/UpdateViewProvider.php]: Is it set, that we do not add docblock to private/protected methods?

File: src/bundle/Controller/UserSettingsController.php [hunk 7]
```diff
@@ -96,21 +96,18 @@ public function listAction(int $page = 1): ListView
     }
 
     /**
+     * @param \EzSystems\EzPlatformUser\View\UserSettings\UpdateView $view
```
Comment (ViniTou) [inline src/bundle/Controller/UserSettingsController.php]: Order could be changed to match arguments

File: src/bundle/Controller/UserSettingsController.php [hunk 8]
```diff
@@ -96,21 +96,18 @@ public function listAction(int $page = 1): ListView
...
+    public function updateAction(Request $request, UpdateView $view): UpdateView
     {
-        $userSetting = $this->userSettingService->getUserSetting($identifier);
-        $data = new UserSettingUpdateData($identifier, $userSetting->value);
+        $userSetting = $view->getUserSetting();
 
-        $form = $this->formFactory->updateUserSetting($identifier, $data);
+        $data = new UserSettingUpdateData($userSetting->identifier, $userSetting->value);
+
+        $form = $this->formFactory->updateUserSetting($userSetting->identifier, $data);
```
Comment (adamwojs) [inline src/bundle/Controller/UserSettingsController.php]: I'm not sure about one thing: does form should be provided by `UpdateViewBuilder` as well?

**Comment by adamwojs** [comment]
PR updated according to @mikadamczyk  @ViniTou suggestions.

**Pull request merged**